File: lecshop_base_modules/lecshop_customer/src/main/java/com/lecshop/openstore/bean/StoreInfo.java
package com.lecshop.openstore.bean;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.lecshop.util.CustomLocalDateTimeDeserializer;
import com.lecshop.util.CustomLocalDateTimeSerializer;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 店铺信息实体类
 *
 * @author sunluyang on 2017/6/13.
 */
@Data
public class StoreInfo {
    /**
     * 主键id
     */
    private long id;
    /**
     * 店铺名称
     */
    private String storeName;
    /**
     * 店铺状态： 0 填写资料中 1 店铺审核中 2 审核通过 3 审核不通过
     */
    private String status;
    /**
     * 公司名称
     */
    private String companyName;
    /**
     * 公司地址
     */
    private String companyAddress;
    /**
     * 公司电话
     */
    private String companyPhone;
    /**
     * 公司邮箱
     */
    private String companyEmail;
    /**
     * 联系人
     */
    private String contactPerson;
    /**
     * 联系人电话
     */
    private String contactPhone;
    /**
     * 法定代表人
     */
    private String legalPerson;
    /**
     * 身份证号码
     */
    private String cardNo;
    /**
     * 身份证照片
     */
    private String carPic;
    /**
     * 营业执照号
     */
    private String busLicense;
    /**
     * 营业执照图片
     */
    private String busLicensePic;
    /**
     * 组织机构代码图片
     */
    private String orgPic;
    /**
     * 税务登记图片
     */
    private String taxPic;
    /**
     * 银行开户名
     */
    private String bankUserName;
    /**
     * 公司银行账号
     */
    private String bankAccount;
    /**
     * 开户银行支行名称
     */
    private String bankName;
    /**
     * 支行银行号
     */
    private String bankNumber;
    /**
     * 开户银行所在地
     */
    private String bankAddress;
    /**
     * 开户银行许可证电子版
     */
    private String bankPic;
    /**
     * 结算周期
     */
    private String billingCycle;
    /**
     * 删除标记 0 未删除 1 删除 默认0
     */
    private String delFlag;
    /**
     * 0三证合一 1 三证不合一
     */
    private String isMerge;
    /**
     * 客服QQ
     */
    private String serviceQQ;
    /**
     * 拒绝原因
     */
    private String reason;
    /**
     * 店铺有效期
     */
    @JsonSerialize(using = CustomLocalDateTimeSerializer.class)
    @JsonDeserialize(using = CustomLocalDateTimeDeserializer.class)
    private LocalDateTime effectiveTime = LocalDateTime.now().plusYears(1);
    /**
     * 创建时间
     */
    @JsonSerialize(using = CustomLocalDateTimeSerializer.class)
    @JsonDeserialize(using = CustomLocalDateTimeDeserializer.class)
    private LocalDateTime createTime = LocalDateTime.now();
    /**
     * 修改时间
     */
    @JsonSerialize(using = CustomLocalDateTimeSerializer.class)
    @JsonDeserialize(using = CustomLocalDateTimeDeserializer.class)
    private LocalDateTime modifyTime;
    /**
     * 删除时间
     */
    @JsonSerialize(using = CustomLocalDateTimeSerializer.class)
    @JsonDeserialize(using = CustomLocalDateTimeDeserializer.class)
    private LocalDateTime delTime;

    public StoreInfo getStoreInfoToEditStoreName(long storeId, String status, String storeName) {
        this.id = storeId;
        this.status = status;
        this.storeName = storeName;
        return this;
    }
}

File: lecshop_api_modules/lecshop_web_store/src/main/webapp/res/js/storemanage/storeinfo.js
// 基本路径
var basePath = $("#basePath").val();
// 初始化列表
var initDataTable = function () {
    LSFetch({
        url: basePath + '/store_mystoreinfo.htm',
        success: function (data) {
            $("p[name='storeId']").text(data.id);
            $("p[name='storeName']").text(data.storeName);
            $("input[name='serviceQQ']").val(data.serviceQQ);
            $("p[name='companyName']").text(data.companyName);
            $("p[name='companyAddress']").text(data.companyAddress);
            $("input[name='companyPhone']").val(data.companyPhone);
            $("p[name='effectiveTime']").text(data.effectiveTime);
            $("input[name='busLicense']").val(data.busLicense);
            $("input[name='contactPerson']").val(data.contactPerson);
            $("input[name='contactPhone']").val(data.contactPhone);
            $("input[name='legalPerson']").val(data.legalPerson);
            $("input[name='cardNo']").val(data.cardNo);
            if (data.taxPic != null && data.taxPic != '') {
                $("img[name='taxPic']").attr("src", data.taxPic);
            }
            if (data.carPic != null && data.carPic != '') {
                $("img[name='carPic']").attr("src", data.carPic);
            }
            if (data.busLicensePic != null && data.busLicensePic != '') {
                $("img[name='busLicensePic']").attr("src", data.busLicensePic);
            }
            if (data.orgPic != null && data.orgPic != '') {
                $("img[name='orgPic']").attr("src", data.orgPic);
            }
            $("input[name='bankUserName']").val(data.bankUserName);
            $("input[name='bankAccount']").val(data.bankAccount);
            $("input[name='bankAddress']").val(data.bankAddress);
            $("input[name='bankName']").val(data.bankName);
            $("input[name='bankNumber']").val(data.bankNumber);
            if (data.bankPic != null && data.bankPic != '') {
                $("img[name='bankPic']").attr("src", data.bankPic);
            }
        }
    });
};

// 初始化列表
initDataTable();

function saveBtn(flag) {
    var isValid = false;
    if (flag == 2) {
        if ($("img[name='carPic']").attr("src") != basePath + "/res/img/no_img.png") {
            $("input[inputName='carPic']").removeClass("required");
        }
        if ($("img[name='busLicensePic']").attr("src") != basePath + "/res/img/no_img.png") {
            $("input[inputName='busLicensePic']").removeClass("required");
        }
        isValid = !$("#carPic").valid() | !$("#busLicensePic").valid()
    }
    if (flag == 3) {
        if ($("img[name='bankPic']").attr("src") != basePath + "/res/img/no_img.png") {
            $("input[inputName='bankPic']").removeClass("required");
        }
        isValid = !$("#bankPic").valid();
    }
    if (!$("#form" + flag).valid() | isValid) {
        return;
    }
    var storeInfo = {
        id: $("p[name='storeId']").text(),
        storeName: $("p[name='storeName']").text(),
        serviceQQ: $("input[name='serviceQQ']").val(),
        companyName: $("p[name='companyName']").text(),
        companyAddress: $("p[name='companyAddress']").text(),
        companyPhone: $("input[name='companyPhone']").val(),
        effectiveTime: $("p[name='effectiveTime']").text(),
        busLicense: $("input[name='busLicense']").val(),
        contactPerson: $("input[name='contactPerson']").val(),
        contactPhone: $("input[name='contactPhone']").val(),
        legalPerson: $("input[name='legalPerson']").val(),
        cardNo: $("input[name='cardNo']").val(),
        bankUserName: $("input[name='bankUserName']").val(),
        bankAccount: $("input[name='bankAccount']").val(),
        bankAddress: $("input[name='bankAddress']").val(),
        bankName: $("input[name='bankName']").val(),
        bankNumber: $("input[name='bankNumber']").val(),
        taxPic: $("img[name='taxPic']").attr("src"),
        carPic: $("img[name='carPic']").attr("src"),
        busLicensePic: $("img[name='busLicensePic']").attr("src"),
        orgPic: $("img[name='orgPic']").attr("src"),
        bankPic: $("img[name='bankPic']").attr("src")
    };
    LSFetch({
        url: basePath + '/store_editmystoreinfo.htm?flag=' + flag,
        data: JSON.stringify(storeInfo),
        contentType: "application/json;charset=utf-8",
        success: function (data) {
            if (data == 1) {
                initDataTable();
                $('#success_dialog').modal('show');
            }
        }
    });
}

// 上传商品图片
function uploadPic(obj) {
    $.ajax({
        url: 'store_uploadtoupyun.htm',
        type: 'POST',
        cache: false,
        data: new FormData($('#' + obj)[0]),
        processData: false,
        contentType: false
    }).done(function (res) {
        $('#' + obj).validate().resetForm();
        $("img[name='" + obj + "']").attr("src", res);
    }).fail(function (res) {
    });
}
